pkg/apis/eksctl.io/v1alpha4/types.go

```diff
@@ -412,6 +432,8 @@ type NodeGroup struct {
 	PreBootstrapCommands []string `json:"preBootstrapCommand,omitempty"`
 	// +optional
 	OverrideBootstrapCommand *string `json:"overrideBootstrapCommand,omitempty"`
+	// +optional
+	OverrideUserData *string `json:"overrideUserData,omitempty"`
```

Comment:
I think this would be a separate feature to discuss. So far, we have stayed away from exposing this...

Reply:
I require this, because the current content of user data is over-complicated and not usable when exported

Reply:
this is the minimal output you need:
```
      UserData:
        Fn::Base64:
          !Sub |
            #!/bin/bash
            set -o xtrace
            /etc/eks/bootstrap.sh ${ClusterName}
            /opt/aws/bin/cfn-signal --exit-code $? \
                     --stack  ${AWS::StackName} \
                     --resource NodeGroup  \
                     --region ${AWS::Region}
```